pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/rebalance/DefaultRebalanceSegmentStrategy.java

```diff
@@ -189,6 +191,10 @@ public IdealState apply(@Nullable IdealState idealState) {
         rebalanceIdealState(idealState, tableConfig, targetNumReplicas, rebalanceUserConfig,
             newPartitionAssignment);
         idealState.setReplicas(Integer.toString(targetNumReplicas));
+        for (Map.Entry<String, Map<String, String>> entry : idealState.getRecord().getMapFields().entrySet()) {
```

Comment:
what if the update fails but the rebalanceIdealState is already updated. We should update the rebalanceIdealState only update to ZK was successful

Reply:
HelixHelper.updateIdealState does not return any indication that it succeeded

Reply:
That is exactly what is happening. But how do I send it back from rebalanceAndUpdateIdealState? It needs to be sent back to be displayed

Reply:
Maybe we can create a blocking retry policy?